### PR TITLE
feat: improved data usage and calculations on render

### PR DESCRIPTION
**Pull request by twjasa**

### CODE REVIEW

File: projects/olympics/index.js [hunk 1]
```diff
@@ -7,23 +7,21 @@ import { useTracking } from "analytics/context";
 import { BarChart } from "./BarChart";
 import PoleVis from "./PoleVis";
 import CustomSelect from "@components/select/select";
-
+import { getOlympicDataToUse } from "./utils";
```
Comment (sikolio) [inline projects/olympics/index.js:10]: This still needs to load all the data, we should just trim the file to contain the relevant data (ie all of the countries who were at the top at some point, but we don't need the data for Chile for example)

File: projects/olympics/BarChart.jsx [hunk 2]
```diff
@@ -129,7 +127,15 @@ export const BarChart = ({ yTitle = "COUNTRY", data, values }) => {
...
-
+                    const xScalePreviousPercent = xScale(previousPercentage) / 6;
+                    const xAnimation =
+                      previousPercentage > 0
+                        ? [MARGIN.LEFT + initialX, MARGIN.LEFT + xScalePreviousPercent]
+                        : MARGIN.LEFT;
+                    const xScaleRowValProperty = xScale(row[val.property]);
+                    const widthMotion = xScaleRowValProperty / 6;
+                    const widthAnimation = [initialWidth, widthMotion];
+                    const yAnimated = LABELS[idx] / 10 + barIdx * BAR_HEIGHT + 17 / 2;
```
Comment (sikolio) [inline projects/olympics/BarChart.jsx:138]: Could we remove the `LABELS` need and move to generate that array of values (or the value for each row itself) from the data that we passed?

File: projects/olympics/index.js (hunk 1 above)
Comment (twjasa) [reply projects/olympics/index.js:10]: 🤔  You mean like actually remove the countries that never reached the top 11? 

And yes, it load all the data but does a single slice to every year once the component mounted and not on every render. 

at the end the data that loads it's max 11 country for every year.
image

this is what you meant? @sikolio

File: projects/olympics/BarChart.jsx (hunk 2 above)
Comment (twjasa) [reply projects/olympics/BarChart.jsx]: Good catch there. Fixed.

**Review by sikolio — APPROVED** [review]

**Pull request merged**